### PR TITLE
fix(github): fix goreleaser configuration to comply with version 2

### PR DESCRIPTION
**Pull request by cderici**
## Description

It looks like a while ago dependabot changed the `uses: goreleaser/goreleaser-action@v5` -> `uses: goreleaser/goreleaser-action@v6`, which apparently uses `goreleaser v2`. And because of that the [release github action is failing](https://github.com/juju/terraform-provider-juju/actions/runs/10045178942) on the `v0.13.0` release.

This updates the goreleaser configuration based on the [deprecation notices for v2](https://goreleaser.com/deprecations/).

Basically the main change is the [changelog skip](https://goreleaser.com/deprecations/#changelogskip).

## Type of change

- Maintenance work (repository related, like Github actions, or revving the Go version, etc.)

## QA steps

Manual QA steps should be done to test this PR.

```
 $ snap install goreleaser --classic
 $ goreleaser check
```

### CODE REVIEW

**Review by hmlanigan — APPROVED** [review]
Please update for this one while we're looking at it:  https://goreleaser.com/deprecations/#builds-for-darwinarm64

You can replace the Darwin/386 with Darwin/arm64, line 35 and remove the commented lines of 27, 28.
